care/facility/api/serializers/patient_consultation.py

```diff
@@ -547,14 +552,20 @@ def validate_create_symptoms(self, value):
                 raise ValidationError(
                     {"onset_date": "This field is required for all symptoms"}
                 )
+
             if obj["onset_date"] > current_time:
                 raise ValidationError(
                     {"onset_date": "Onset date cannot be in the future"}
                 )
-            if "cure_date" in obj and obj["cure_date"] < obj["onset_date"]:
-                raise ValidationError(
-                    {"cure_date": "Cure date should be after onset date"}
-                )
+
+            if "cure_date" in obj:
+                if obj["cure_date"] < obj["onset_date"]:
+                    raise ValidationError(
+                        {"cure_date": "Cure date should be after onset date"}
+                    )
+                obj["clinical_impression_status"] = ClinicalImpressionStatus.COMPLETED
+            else:
+                obj["clinical_impression_status"] = ClinicalImpressionStatus.IN_PROGRESS
```

Comment:
"cure_date" may or may not be sent from client as `null` in the obj.
This would cause incorrect results right?

<img width="533" alt="image" src="https://github.com/coronasafe/care/assets/25143503/e3f01052-a3e9-4789-bd50-c085d2e00142">